discretisedfield/field.py

```diff
@@ -1383,7 +1460,8 @@ def __matmul__(self, other):
                        f'and {other.dim=} fields.')
                 raise ValueError(msg)
         elif isinstance(other, (tuple, list, np.ndarray)):
-            return self @ self.__class__(self.mesh, dim=3, value=other)
+            return self @ self.__class__(self.mesh, dim=3, value=other,
+                                         components=self.components)
```

Comment:
This could think of making a partial function for avoiding this repetition for so many different methods. No need to do it now, we can do that when we refactor.